teps/0135-coscheduling-pipelinerun-pods.md

```diff
@@ -137,12 +138,79 @@ PipelineRun pods can be scheduled to the same node using a mechanism similar to
-- "disabled": The default value.
-- "coschedule-workspaces": The existing affinity assistant behavior. This option runs all of a PipelineRun's pods sharing the same workspace on the same node.
+- "coschedule-workspaces": The default value. The existing affinity assistant behavior. This option runs all of a PipelineRun's pods sharing the same workspace on the same node.
+- "disabled": Pod coscheduling feature is disabled.
 - "coschedule-pipelineruns": This option runs all of a PipelineRun's pods on the same node.
 - "isolate-pipelineruns": This option runs all of a PipelineRun's pods on the same node, and only allows one PipelineRun to run on a node at a time. This option is intended for use in clusters where a cluster autoscaler scales the number of nodes up and down.
 
-If the existing affinity assistant is enabled, the "coscheduling" flag must be set to "disabled". We will deprecate the "disable-affinity-assistant" flag in favor of the "coscheduling" flag, as discussed in ["Upgrade and Migration Strategy"](#upgrade-and-migration-strategy).
+If the existing affinity assistant is enabled, the "coscheduling" flag must be set to "coschedule-workspaces". We will deprecate the "disable-affinity-assistant" flag in favor of the "coscheduling" flag, as discussed in ["Upgrade and Migration Strategy"](#upgrade-and-migration-strategy).
+
+The following chart summarizes the affinity assistant behaviors with different combinations of the "disable-affinity-assistant" and "coscheduling" feature flags during migration (when both feature flags are present) and after the migration (when only the "coscheduling" flag is present):
+
+<table>
+    <thead>
+        <tr>
+            <th>disabled-affinity-assistant</th>
+            <th>coscheduling</th>
+            <th>behavior during migration</th>
+            <th>behavior after migration</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>false (default)</td>
+            <td>disabled</td>
+            <td>N/A: invalid</td>
+            <td>disabled</td>
+        </tr>
+        <tr>
+            <td>false (default)</td>
+            <td>coschedule-workspaces (default)</td>
+            <td>coschedule workspaces</td>
+            <td>coschedule workspaces</td>
+        </tr>
+        <tr>
+            <td>false (default)</td>
+            <td>coschedule-pipelineruns</td>
+            <td>N/A: invalid</td>
+            <td>coschedule pipelineruns</td>
+        </tr>
+        <tr>
+            <td>false (default)</td>
+            <td>isolate-pipelineruns</td>
+            <td>N/A: invalid</td>
+            <td>isolate pipelineruns</td>
+        </tr>
+        <tr>
+            <td>true</td>
+            <td>disabled</td>
+            <td>disabled</td>
+            <td>disabled</td>
+        </tr>
+        <tr>
+            <td>true</td>
+            <td>coschedule-workspaces (default)</td>
+            <td>disabled</td>
```

Comment:
I'm unsure about this combination - in this case, the affinity-assistant flag (disabled) prevails over the coschedule flag (enabled, default). I was expecting this to be "N/A: invalid" like in the flip case of AA enabled and coschedule disabled.

Reply:
I actually perfer this way of thinking better! I summarize this new idea into this chart in a [shared doc](https://docs.google.com/document/d/1ktiBUJPsPV4ZNzwWqsOI1pVVyLQj6uWii0dM7jXK-K8/edit?resourcekey=0-5FhHpJ4P_Y_FzTv_zEnqUg). Please take a look!

@lbernick @vdemeester @pritidesai

Reply:
Sounds good, thank @lbernick - I don't think there is a clear winner between the two, but I find your phrasing really good and useful for docs and release notes. It's ok for me to keep the current plan.

Reply:
Thanks for your inputs! I agree that which one is less confusing is quite personal. In this case, I think it makes sense to keep the current chart. But I will make sure to add Lee's rephrasing to the documentation and implementation PR once the feature is ready.

@afrittoli, do you think you can approve this PR and get it merged?

Reply:
thank you @QuanZhang-William for putting up this PR. I am sorry for jumping in last minute 😞 

The feature flag is named `coscheduling` which sounds odd for a name. Most of the supported values starts with `coschedule-`. 

Has anyone else thought about the naming?

How does `coexist` sound as an alternative with `workspaces`, `all-pipelineruns`, `single-pipelinerun`, etc.

I am not that great with names either but checking if other folks finding this confusing

Reply:
Hi @pritidesai, no worries! Yeah, I agree naming is not an easy thing...

When reading the current feature flag `coscheduling` and the values, I personally didn't find it is too confusing to read, but not sure about others.

If we use `coexist` with `all-pipelineruns`, I feel some people could understand it as we will schedule all the pipelineruns (not pipeline tasks) to the same node. 

I'm definitely open to this discussion, but I think we can defer this topic to another PR, as the purpose of this one is to update the default value to preserve AA behavior before&after the migration ?

Reply:
In either case, I wouldn't mind agreeing on the final naming on the implementation PR.

Reply:
> Maybe we could have some along these lines:
> 
> * flag name: `coschedule`
> * values: `[disabled, workspaces, pipelines, isolate]`
> 
> To avoid the repetition of coschedule?

I think this is a good suggestion, but maybe `pipelineruns` instead of `pipelines`

Reply:
@pritidesai , do you think we can merge this TEP change as it is, and defer the naming discussion to this implementation PR:
https://github.com/tektoncd/pipeline/pull/6790

I can go back and update TEP later if we eventually decide to update the names 😁

Reply:
I am sorry @QuanZhang-William I didn't mean to block this PR for the naming conventions.

I just want to make sure its something not forgotten in a long run. As we heavily rely on the existing feature flag and trying to make sure the transition is smooth for the cluster operators.